File: GroupBar_Visibleitem_Selection_C#/Form1.cs
using Syncfusion.Windows.Forms;
using Syncfusion.Windows.Forms.Tools;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GroupBar_selection
{
    public partial class Form1 : MetroForm
    {
        GroupBar groupBar1 = new GroupBar();
        public Form1()
        {
            InitializeComponent();
            groupBar1.Size = new Size(220, 300);
            groupBar1.VisualStyle = VisualStyle.Metro;
            groupBar1.BackColor = System.Drawing.Color.FromArgb(((int)(((byte)(235)))), ((int)(((byte)(235)))), ((int)(((byte)(235)))));

            groupBar1.GroupBarItemSelected += GroupBar1_GroupBarItemSelected;

            GroupBarItem groupBarItem1 = new GroupBarItem();
            groupBarItem1.Text = "Mail";

            GroupBarItem groupBarItem2 = new GroupBarItem();
            groupBarItem2.Text = "Contacts";
            
            GroupBarItem groupBarItem3 = new GroupBarItem();
            groupBarItem3.Text = "Tasks";
            groupBarItem3.Visible = false;
            
            GroupBarItem groupBarItem4 = new GroupBarItem();
            groupBarItem4.Text = "Notes";

            groupBar1.GroupBarItems.AddRange(new GroupBarItem[] { groupBarItem1, groupBarItem2, groupBarItem3, groupBarItem4 });

            this.Controls.Add(groupBar1);
        }

        private void GroupBar1_GroupBarItemSelected(object sender, EventArgs e)
        {
            String mygroup = this.groupBar1.VisibleGroupBarItems[this.groupBar1.SelectedItem].Text;
            MessageBox.Show(mygroup);
        }
    }
}

File: GroupBar_Visibleitem_Selection_VB/GroupBar_Selectitems/Form1.vb
Imports Syncfusion.Windows.Forms
Imports Syncfusion.Windows.Forms.Tools

Partial Public Class Form1
    Inherits MetroForm
    Private groupBar1 As New GroupBar()
    Public Sub New()
        InitializeComponent()
        groupBar1.Size = New Size(220, 300)
        groupBar1.VisualStyle = VisualStyle.Metro
        groupBar1.BackColor = System.Drawing.Color.FromArgb((CInt(Fix((CByte(235))))), (CInt(Fix((CByte(235))))), (CInt(Fix((CByte(235))))))

        AddHandler groupBar1.GroupBarItemSelected, AddressOf GroupBar1_GroupBarItemSelected

        Dim groupBarItem1 As New GroupBarItem()
        groupBarItem1.Text = "Mail"

        Dim groupBarItem2 As New GroupBarItem()
        groupBarItem2.Text = "Contacts"

        Dim groupBarItem3 As New GroupBarItem()
        groupBarItem3.Text = "Tasks"
        groupBarItem3.Visible = False

        Dim groupBarItem4 As New GroupBarItem()
        groupBarItem4.Text = "Notes"

        groupBar1.GroupBarItems.AddRange(New GroupBarItem() {groupBarItem1, groupBarItem2, groupBarItem3, groupBarItem4})

        Me.Controls.Add(groupBar1)
    End Sub

    Private Sub GroupBar1_GroupBarItemSelected(ByVal sender As Object, ByVal e As EventArgs)
        Dim mygroup As String = Me.groupBar1.VisibleGroupBarItems(Me.groupBar1.SelectedItem).Text
        MessageBox.Show(mygroup)
    End Sub
End Class